deploy_and_run_integration_tests.sh

```diff
@@ -0,0 +1,187 @@
+  build_file="$(ls cloudbuild.yaml)" || true
+  if [[ ${build_file} != "cloudbuild.yaml" ]]; then
+    usage
+    color_print "ERROR: Cannot find 'cloudbuild.yaml'!" "${RED}"
+    exit 1
+  fi
+}
+
+#################################################
+# Parses arguments and does some sanity checking. Any non-recognized argument is
+# added to ${TEST_ARGUMENTS} to be passed to the test script later.
+# Arguments:
+#   It is expected that this is called with $@ of the main script.
+#################################################
+parse_args() {
+  while [[ $# -gt 0 ]]; do
+    if [[ "$1" = "${KEEP_IMAGE_OPT}" ]]; then
+      keep_image="yes"  # can be any non-empty string
+      shift
+    elif [[ "$1" = "${IMAGE_TAG_OPT}" ]]; then
+      shift
+      if [[ $# == 0 ]]; then
+        usage
+        color_print "ERROR: No tag provided after ${IMAGE_TAG_OPT}!" "${RED}"
+        exit 1
+      fi
+      image_tag="$1"
+      color_print "Using custom image tag: ${image_tag}" "${GREEN}"
+      shift
+    elif [[ "$1" = "${SKIP_BUILD_OPT}" ]]; then
+      skip_build="yes"  # can be any non-empty string
+      shift
+    else
+      TEST_ARGUMENTS="${TEST_ARGUMENTS} $1"
+      shift
+    fi
+  done
+  if [[ ( ! -z ${skip_build} ) && ( -z ${image_tag} ) ]]; then
+    color_print "ERROR: ${SKIP_BUILD_OPT} with no ${IMAGE_TAG_OPT}" "${RED}"
+    exit 1
+  fi
+}
+
+#################################################
+# Deactivates virtualenv and removes its directory.
+#################################################
+clean_up_virtualenv() {
+  color_print "Removing integration test environment ${temp_dir}" "${GREEN}"
+  deactivate
+  rm -rf "${temp_dir}"
+}
+
+#################################################
+# Main
+#################################################
+check_dir
+parse_args $@
+color_print "Arguments to be passed to the test script: " "${GREEN}"
+color_print "${TEST_ARGUMENTS}" "${GREEN}"
+
+if [[ -z "${image_tag}" ]]; then
+  time_tag="$(date +%F-%H-%M-%S)"
+  image_tag="test_${time_tag}"
+fi
+full_image_name="gcr.io/${PROJECT}/gcp-variant-transforms:${image_tag}"
+
+if [[ -z "${skip_build}" ]]; then
+  color_print "Building the Docker image with tag ${image_tag}" "${GREEN}"
+  # TODO(bashir2): This will pick and include all directories in the image,
+  # including local build and library dirs that do not need to be included.
+  # Update this to include only the required files/directories.
+  gcloud container builds submit --config "${build_file}" \
+      --project "${PROJECT}" \
+      --substitutions _CUSTOM_TAG_NAME="${image_tag}" .
+fi
+
+# Running integration tests in a temporary virtualenv
+temp_dir="$(mktemp -d)"
+color_print "Setting up integration test environment in ${temp_dir}" "${GREEN}"
+# Since we have no prompt we need to disable prompt changing in virtualenv.
+export VIRTUAL_ENV_DISABLE_PROMPT="something"
+virtualenv "${temp_dir}"
+source ${temp_dir}/bin/activate;
+trap clean_up_virtualenv EXIT
```

Comment:
do we need to clean up images here?

Reply:
Done.
When Asha suggested to have the trap I thought I leave the image clean-up out because when something goes wrong, it is very likely that we need the image (to test again). But now that you also suggested it, I think it makes more sense to more people to delete it, so I did it :-)